### ISSUE TITLE
NER confidence

### ISSUE THREAD

**Issue by flaviussn** [ThilinaRajapakse/simpletransformers#348]
I need the _word confidence_ from NER prediction and I have tried to use `model_outputs`. I've noticed that the `model_output` contains the _transformers token logits_ and it's impossible to know or combine these logits to obtain _word logits_ or your own input tokenization, outside this function. (I am wrong?)

https://github.com/ThilinaRajapakse/simpletransformers/blob/ebd8dc149677fa775ab3fb2a9d0bf8efe8bf46c2/simpletransformers/ner/ner_model.py#L726

I think that there are two options implementing similar to the [pred list](https://github.com/ThilinaRajapakse/simpletransformers/blob/ebd8dc149677fa775ab3fb2a9d0bf8efe8bf46c2/simpletransformers/ner/ner_model.py#L734) that returns the prediction for each word.

- Create a new get function that gives us access to _word logits_
- Return an extra variable that contains _word logits_ in the predict function.

In a way or another, I think that this option should be provided because is very useful to get prediction confidence. 

What option do you think that suits better? Any better idea? I offer myself to implement it :smile:

**Comment by ThilinaRajapakse** [comment]
Is the current `model_output` returned from the `predict()` function useless outside of itself? (I can't remember what all these different variables in the NER model are)

I think the second option sounds better. As the `predict()` function is supposed to return the "clean" predictions and the "raw" outputs, I don't think we should introduce a new method to do something similar.

However, if the answer to the first question is yes (useless), then maybe we can just replace the current `model_output` with your proposed output? Otherwise, we could add it as a third return value, which can be returned using an optional flag (to make sure there is no breaking change).

What do you think? (I'd be happy to second your nomination of yourself! :wink: )

**Comment by flaviussn** [comment]
@ThilinaRajapakse I think that the current `model_output` is useless mainly because outside the function you can't correlate the `logits` (model_output) with the `tokens`  that are extracted inside and not returned. 

Maybe we can return the `model_output` as a dictionary of words, and each word value contains a list of logits that correspond to the word tokens. I mean:

```python
internal_bert_tokens =  ['simple', '##tra', '##ns', '##former', '##s', 'rocks']
preds = [[{'simpletransformers': 'O'}, {'rocks': 'O'}]]
model_output = [[{'simpletransformers': [logits_simple,logits_tra,logits_ns,logits_former,logits_s]}, {'rocks': [logits_rocks]}]]
```

What do you think?

**Comment by ThilinaRajapakse** [comment]
I see.

Yeah, that looks great. I think we should replace the current `model_output` with your suggestion.
